src/test/java/org/jenkinsci/plugins/workflow/libs/LibraryAdderTest.java

```diff
@@ -293,4 +294,22 @@ public static void register(Object o) {
         }
     }
 
+    @Issue({"JENKINS-38021", "JENKINS-31484"})
+    @Test public void gettersAndSetters() throws Exception {
+        sampleRepo.init();
+        sampleRepo.write("vars/config.groovy", "class config implements Serializable {private String foo; public String getFoo() {return(/loaded ${this.foo}/)}; public void setFoo(String value) {this.foo = value.toUpperCase()}}");
+        sampleRepo.git("add", "vars");
+        sampleRepo.git("commit", "--message=init");
+        GlobalLibraries.get().setLibraries(Collections.singletonList(
+            new LibraryConfiguration("config",
+                new SCMSourceRetriever(new GitSCMSource(null, sampleRepo.toString(), "", "*", "", true)))));
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("@Library('config@master') _; timeout(1) {config.foo = 'bar'; echo(/set to $config.foo/)}", false));
```

Comment:
`timeout` helped to test effectiveness of https://github.com/jenkinsci/workflow-cps-plugin/pull/101.